Sources/Networking/HTTPClient/HTTPClient.swift

```diff
@@ -582,6 +624,87 @@ private extension HTTPClient {
 
 }
 
+// MARK: - Request Retry Logic
+extension HTTPClient {
+    internal func retryRequestIfNeeded(
+        request: HTTPClient.Request,
+        httpURLResponse: HTTPURLResponse?
+    ) {
+        // retryCount is incremented before the retry is executed, so don't stop retries if
+        // retryCount == self.retryOptions.maxNumberOfRetries
+        guard request.retryCount > self.retryOptions.maxNumberOfRetries else { return }
+        guard let httpURLResponse = httpURLResponse else { return }
+        guard shouldRetryRequest(withStatusCode: httpURLResponse.httpStatusCode) else { return }
```

Comment:
I'd probably make this the first check since conceptually it seems more important than the others

Reply:
we're always returning on these, maybe we could just chain them with `&&` and `,`?

Reply:
then again we may wanna log, especially for the case where we've reached max retries on 429

Reply:
Logging here when we've reached the max retry count is a great idea - I'll add one in and combine the other two guard statements.